### ISSUE TITLE
Editing an update clears the package list, can't view update

### ISSUE THREAD

**Issue by mooninite** [fedora-infra/bodhi#202]
I needed to edit an existing update that was pending to switch to a newer build.

After adding the newer package NVR to the update and clicking "edit" the page took me to a list of all updates. When I returned to my profile page the update line exists, but the package name has been wiped out and I cannot click on anything.

Screenshots
After editing the mediawiki-1.25.2-1.fc23 update to mediawiki-1.25.2-2.fc23: http://i.imgur.com/uYfqD0R.png
After editing the mediawiki-1.25.2-1.fc22 update to mediawiki-1.25.2-2.fc22: http://i.imgur.com/7fQJAc7.png

**Comment by lmacken** [comment]
```
[Thu Aug 20 04:03:34.876486 2015] [:error] [pid 21940] 2015-08-20 04:03:34,876 DEBUG [bodhi][Dummy-1] validated = {'close_bugs': True, 'stable_karma': 2, 'csrf_token': u'<removed>', 'requirements': None, 'builds': [], 'autokarma': True, 'edited': u'mediawiki-1.25.2-1.fc23', 'suggest': <unspecified>, 'notes': u'* (T94116) SECURITY: Compare API watchlist token in constant time\\n* (T97391) SECURITY: Escape error message strings in thumb.php\\n* (T106893) SECURITY: Don\\'t leak autoblocked IP addresses on Special:DeletedContributions\\n* (T102562) Fix InstantCommons parameters to handle the new HTTPS-only policy of Wikimedia Commons.\\n* (T100767) Setting a configuration setting for skin or extension to false in LocalSettings.php was not working.\\n* (T100635) API action=opensearch json output no longer breaks when $wgDebugToolbar is enabled.\\n* (T102522) Using an extension.json or skin.json file which has a "manifest_version" property for 1.26 compatability will no longer trigger warnings.\\n* (T86156) Running updateSearchIndex.php will not throw an error as page_restrictions has been added to the locked table list.\\n* Special:Version would throw notices if using SVN due to an incorrectly named variable. Add an additional check that an index is defined.\\n\\n', 'request': <testing>, 'bugs': [1253281, 1253280, 1252194], 'unstable_karma': -2, 'require_bugs': True, 'require_testcases': True, 'type': <security>, 'severity': <unspecified>}
[Thu Aug 20 04:03:34.876731 2015] [:error] [pid 21940] 2015-08-20 04:03:34,876 INFO  [bodhi][Dummy-1] Editing update: mediawiki-1.25.2-1.fc23
[Thu Aug 20 04:03:38.281694 2015] [:error] [pid 21940] 2015-08-20 04:03:38,281 INFO  [bodhi][Dummy-1] Removing f23-updates-candidate tag from mediawiki-1.25.2-1.fc23
[Thu Aug 20 04:03:38.419168 2015] [:error] [pid 21940] 2015-08-20 04:03:38,418 INFO  [bodhi][Dummy-1] Removing f23-updates-testing-pending tag from mediawiki-1.25.2-1.fc23
```

It looks like it made it past the validators with an empty list of builds...

**Comment by lmacken** [comment]
Just to confirm, you pasted the nvr into the 'Candidate builds' field, hit enter, checked the box next to the new build and unchecked the old build? or did you use the package field to auto-complete the candidate builds?

**Comment by pghmcfc** [comment]
Just had this happen to me with my F23 perl-Test-Vars update.
I pasted the nvr into the "Candidate builds" field, did *not* hit enter as I didn't see any guidance to do so, unchecked the old build and clicked on the "Edit" button at the bottom of the screen. So I appear to have deleted all builds from my update and cannot now access it to add the correct build back in.

**Comment by pghmcfc** [comment]
Submitted a new update successfully after pressing enter after the nvr.
Managed to locate my original update (by sequencing through FEDORA-2015-nnnnn update IDs), which now has no builds: https://bodhi.fedoraproject.org/updates/FEDORA-2015-13793
I'd like to delete that update now, but can't see how.

**Comment by mooninite** [comment]
I did not press enter. It was not clear that it was required. I copy & pasted the new nvr into the candidate build box and clicked on the edit button.

**Comment by rdieter** [comment]
In my case, I don't recall hitting enter either

**Comment by lmacken** [comment]
So yeah, there are two parts to this issue:

 - Fix the "Candidate builds" UI element behavior
 - Fix the validators on the backend to ensure updates cannot be edited with an empty build list.

**Comment by lmacken** [comment]
Fixed in git, going to prod shortly. The UI fix RFE is filed as #249
